toolchain/sem_ir/inst_fingerprinter.cpp

```diff
@@ -416,10 +417,19 @@ struct Worklist {
       if (!std::holds_alternative<InstId>(next)) {
         // Add the contents of the `next` instruction so they all contribute to
         // the `contents`.
-        if (auto* impl_id = std::get_if<ImplId>(&next)) {
-          Add(*impl_id);
-        } else if (auto* inst_block_id = std::get_if<InstBlockId>(&next)) {
-          Add(*inst_block_id);
+        CARBON_KIND_SWITCH(next) {
+          case CARBON_KIND(InstId _):
+            // InstId is checked for above.
+            CARBON_FATAL();
```

Comment:
I think it's a good habit to include strings for check messages, and it seems like the comment can be used here:

```suggestion
            CARBON_FATAL("InstId is checked for above.");
```

Reply:
Ah, I had that at first and then thought it was a weird crash message and changed it :P Sounds good anyway